File: Pokedex/iosApp/iosApp/Util/ImageUtil.swift
//
//  ImageUtil.swift
//  iosApp
//
//  Created by neyogiry on 1/12/24.
//  Copyright © 2024 orgName. All rights reserved.
//

final class ImageUtil {
    static func imageURL(from url: String) -> String {
        let baseImageUrl = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/%@.png"
        let array = url.components(separatedBy: "/")
        let id = array[array.count-2]
        return String(format: baseImageUrl, arguments: [id])
    }
}

File: Pokedex/composeApp/src/androidMain/kotlin/com/neyogiry/kmm/sample/pokedex/util/ImageUtil.kt
package com.neyogiry.kmm.sample.pokedex.util

object ImageUtil {

    fun pokemonImageUrl(url: String): String {
        val index = url.split("/".toRegex()).dropLast(1).last()
        return "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/$index.png"
    }

}